public/components/panel/workstationPanel/workstationPanel.js

```diff
@@ -41,30 +73,66 @@ const WorkstationPanel = props => {
 
   /**
    * @param {Number} index
-   * @param {Number} newEditedCol new position on grid
+   * @param {Number} newAudioDelay new position on grid
    */
-  const handleGridSampleDragEnd = (index, newEditedCol) => {
+  const handleGridSampleDragEnd = (index, newAudioDelay) => {
     if (index < 0 || index >= props.loadedSampleList.length) {
       return
     }
     let newValue = getLoadedSampleCopy()
-    newValue[index].sampleColIndex = newEditedCol
+    newValue[index].sampleAudioDelay = newAudioDelay
+    fixResizeOverCorrections(newValue[index])
     props.setLoadedSampleList(newValue)
   }
 
   /**
    * @param {Number} index
-   * @param {Number} lengthDelta change in item length. can be +/-
+   * @param {Number} durationDelta change in item length. can be +/-
+   * @param {ResizeDirection} direction
    */
-  const handleGridSampleResizeEnd = (index, lengthDelta) => {
+  const handleGridSampleResizeEnd = (index, durationDelta, direction) => {
     if (index < 0 || index >= props.loadedSampleList.length) {
       return
     }
     let newValue = getLoadedSampleCopy()
-    newValue[index].samplePlayLength += lengthDelta
+    let editedSample = newValue[index]
+
+    if ('right' === direction) {
```

Comment:
why is this set to "right"?

Reply:
Cropping from the left or right have different effects. When you crop from the right, you are increasing the delay (how long to wait before playing something), increasing start (what point in the sample to start playing), and decreasing the duration (how long to play x sample for). When you crop from the left, you're just decreasing the duration.